src/prefect/orion/api/block_documents.py

```diff
@@ -125,6 +125,32 @@ async def update_block_document_data(
                 session=session,
                 block_document_id=block_document_id,
                 block_document=block_document,
+                merge_existing_data=True,
+            )
+    except ValueError:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST)
+
+    if not result:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND, detail="Block document not found"
+        )
+
+
+@router.put("/{id:uuid}", status_code=status.HTTP_204_NO_CONTENT)
```

Comment:
This is not the correct use of `PUT`, which should be "checking if resource exists then update, else create new resource".

Reply:
Would using `POST` for this operation be more appropriate?